packages/medusa/src/api/routes/admin/workflows/index.ts

```diff
@@ -0,0 +1,74 @@
+import middlewares, {
+  transformBody,
+  transformQuery,
+} from "../../../middlewares"
+
+import { Router } from "express"
+import {
+  allowedAdminWorkflowExecutionsRelations,
+  defaultAdminWorkflowExecutionDetailFields,
+  defaultAdminWorkflowExecutionsFields,
+  defaultAdminWorkflowExecutionsRelations,
+} from "./query-config"
+import {
+  AdminGetWorkflowExecutionDetailsParams,
+  AdminGetWorkflowExecutionsParams,
+  AdminPostWorkflowsAsyncResponseReq,
+  AdminPostWorkflowsRunReq,
+} from "./validators"
+
+const route = Router()
+
+const retrieveTransformQueryConfig = {
+  defaultFields: defaultAdminWorkflowExecutionDetailFields,
+  defaultRelations: defaultAdminWorkflowExecutionsRelations,
+  allowedRelations: allowedAdminWorkflowExecutionsRelations,
+  isList: false,
+}
+
+const listTransformQueryConfig = {
+  ...retrieveTransformQueryConfig,
+  defaultFields: defaultAdminWorkflowExecutionsFields,
+  isList: true,
+}
+
+export default (app) => {
+  app.use("/workflows", route)
```

Comment:
thought(non-blocking): Should we call this route `/workflow-executions` instead, knowing that we will have both a Workflows and Executions overview in the Admin in the future?

So the routes become:

```
GET /admin/workflow-executions

GET /admin/workflow-executions/:id
{ transaction_id: "..." }

POST /admin/workflow-executions/:id/run

POST /admin/workflow-executions/:id/steps/:step_id/success
{ transaction_id: "..." }

POST /admin/workflow-executions/:id/steps/:step_id/failure
{ transaction_id: "..." }

```

Reply:
@carlos-r-l-rodrigues, what do you think of this change? 

I know we don't expose all existing workflows + definitions yet, but I think we will some day.

Reply:
That makes sense. And executions can also be fetched by their id. Hence the `idOrObject`.
I am fixing that.

Reply:
Cool, sounds good 👍